project/Build.scala

```diff
@@ -1419,4 +1424,85 @@ object Build extends sbt.Build {
           }
       )
   ).dependsOn(partest % "test", library)
+
+  lazy val scalaTestSuite: Project = Project(
+    id = "scalaTestSuite",
+    base = file("scala-test-suite"),
+    settings = commonSettings ++ myScalaJSSettings ++ Seq(
+      publishArtifact in Compile := false,
+
+      testOptions += Tests.Argument(TestFrameworks.JUnit, "-v", "-a"),
+
+      unmanagedSources in Test ++= {
+        def loadList(listName: String): Set[String] = {
+          val listsDir = (resourceDirectory in Test).value / scalaVersion.value
+          val buff = scala.io.Source.fromFile(listsDir / listName)
+          val lines = buff.getLines().collect {
+            case line if !line.startsWith("#") && line.nonEmpty => line
+          }
+          lines.toSet
+        }
+
+        val blackList: Set[String] = loadList("BlacklistedTests.txt")
+        val whiteList: Set[String] = loadList("WhitelistedTests.txt")
+
+        val jUnitTestsPath =
+          (artifactPath in fetchScalaSource in partest).value / "test" / "junit"
+
+        def listFiles(file: File, acc: List[File], listed: List[String],
+            listedErr: List[String]): (List[File], List[String], List[String]) = {
+          if (file.isDirectory) {
+            file.listFiles().foldLeft((acc, listed, listedErr)) { (foldAcc, f) =>
+              listFiles(f, foldAcc._1, foldAcc._2, foldAcc._3)
+            }
```

Comment:
Use `(topLevelDir ** "*.scala").get` to get all files recursively from `topLevelDir`. Don't hand-roll your own directory recursion.